pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -233,13 +227,17 @@ def configure_progress_bar(
                     "Trainer was configured with `enable_progress_bar=False`"
                     f" but found `{progress_bar_callback.__class__.__name__}` in callbacks list."
                 )
-        elif refresh_rate > 0 and enable_progress_bar:
+        # Setting refresh_rate to 0 disables the progress bar callback, so return early
+        if refresh_rate == 0:
+            return
+        if os.getenv("COLAB_GPU") and refresh_rate is None:
+            # smaller refresh rate on colab causes crashes, choose a higher value
+            refresh_rate = 20
+        refresh_rate = 1 if refresh_rate is None else refresh_rate
+
```

Comment:
this is minor refactoring to make the failure cases/early exits happen first to avoid extra work